#!/usr/bin/env python
# human pose prediction configuration

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, bool_t

gen = ParameterGenerator()

gen.add("velscale_lower", double_t, 0,
        "human slow-down velocity multiplier for velcity-scale calculation", 0.8, 0.0, 1.0)
gen.add("velscale_nominal", double_t, 0,
        "human nominal velocity multiplier for velcity-scale calculation", 1.0, 0.0, 10.0)
gen.add("velscale_higher", double_t, 0,
        "human speed-up velocity multiplier for velcity-scale calculation", 1.2, 1.0, 10.0)
gen.add("velscale_angle", double_t, 0,
        "deviation angle for human position predictions for velcity-scale calculation", 0.1, 0.0, 3.14)
gen.add("velscale_mul", double_t, 0,
        "multiplier for human velocities for velocity-scale calculation", 1.0, 0.001, 10.0)
gen.add("velobs_mul", double_t, 0,
        "multiplier for human velocities for velocity-obstacle calculation", 1.0, 0.001, 10.0)

gen.add("velobs_min_rad", double_t, 0,
        "minimum radius for velcoity-obstacle calculation", 0.25, 0.0, 10)
gen.add("velobs_max_rad", double_t, 0,
        "maxium radius for volocity-obstacle calculation", 0.75, 0.0, 10)
gen.add("velobs_max_rad_time", double_t, 0,
        "time for maximum radius for velocity-obstacle calculation", 4.0, 0.0, 60.0)
gen.add("velobs_use_ang", bool_t, 0,
        "wheter to use angular velocity for velocity-obstacle calculation", True)

exit(gen.generate("human_path_prediction",
     "human_path_prediction", "HumanPathPrediction"))
